public/app/plugins/datasource/elasticsearch/dataquery.cue

```diff
@@ -0,0 +1,424 @@
+								unit?: string
+								mode?: string
+							}
+						} @cuetsy(kind="interface")
+
+						#BasePipelineMetricAggregation: {
+							#MetricAggregationWithField
+							type:         #PipelineMetricAggregationType
+							pipelineAgg?: string
+						} @cuetsy(kind="interface")
+
+						#PipelineMetricAggregationWithMultipleBucketPaths: {
+							#BaseMetricAggregation
+							type: #PipelineMetricAggregationType
+							pipelineVariables?: [...#PipelineVariable]
+						} @cuetsy(kind="interface")
+
+						#MovingAverageModel: "simple" | "linear" | "ewma" | "holt" | "holt_winters" @cuetsy(kind="type")
+
+						#MovingAverageModelOption: {
+							label: string
+							value: #MovingAverageModel
+						} @cuetsy(kind="interface")
+
+						#BaseMovingAverageModelSettings: {
+							model:   #MovingAverageModel
+							window:  string
+							predict: string
+						} @cuetsy(kind="interface")
+
+						#MovingAverageSimpleModelSettings: {
+							#BaseMovingAverageModelSettings
+							model: #MovingAverageModel & "simple"
+						} @cuetsy(kind="interface")
+
+						#MovingAverageLinearModelSettings: {
+							#BaseMovingAverageModelSettings
+							model: #MovingAverageModel & "linear"
+						} @cuetsy(kind="interface")
+
+						#MovingAverageEWMAModelSettings: {
+							#BaseMovingAverageModelSettings
+							model: #MovingAverageModel & "ewma"
+							settings?: {
+								alpha?: string
+							}
+							minimize: bool
+						} @cuetsy(kind="interface")
+
+						#MovingAverageHoltModelSettings: {
+							#BaseMovingAverageModelSettings
+							model: #MovingAverageModel & "holt"
+							settings: {
+								alpha?: string
+								beta?:  string
+							}
+							minimize: bool
+						} @cuetsy(kind="interface")
+
+						#MovingAverageHoltWintersModelSettings: {
+							#BaseMovingAverageModelSettings
+							model: #MovingAverageModel & "holt_winters"
+							settings: {
+								alpha?:  string
+								beta?:   string
+								gamma?:  string
+								period?: string
+								pad?:    bool
+							}
+							minimize: bool
+						} @cuetsy(kind="interface")
+
+						// #MovingAverageModelSettings Not sure how to do this one:
+						// export type MovingAverageModelSettings<T extends MovingAverageModel = MovingAverageModel> = Partial<
+						//   Extract<
+						//     | MovingAverageSimpleModelSettings
+						//     | MovingAverageLinearModelSettings
+						//     | MovingAverageEWMAModelSettings
+						//     | MovingAverageHoltModelSettings
+						//     | MovingAverageHoltWintersModelSettings,
+						//     { model: T }
+						//   >
+						// >;
+
+						// #MovingAverage Not sure how to do this one - for now just mocking it:
+						#MovingAverage: {} @cuetsy(kind="interface")
+						// export interface MovingAverage<T extends MovingAverageModel = MovingAverageModel>
+						//   extends BasePipelineMetricAggregation {
+						//   type: 'moving_avg';
+						//   settings?: MovingAverageModelSettings<T>;
+						// }
```

Comment:
@Clarity-89 @sdboyer Also any idea/help on how to define these?

Reply:
I think `MovingAverageModelSettings` should probably be just manually defined in `types.ts`, I don't think cue can represent anything like that. 
`MovingAverage` will need to be modified in `types.ts`, similar to `BasePipelineMetricAggregation`. In cue we can define it as:
```
#MovingAverage: {
   #BasePipelineMetricAggregation
   type: "moving_avg"
   settings?: {...}
 } @cuetsy(kind="interface")
```
The in types.ts:
```typescript
import {
  MovingAverageSimpleModelSettings,
  MovingAverageLinearModelSettings,
  MovingAverageEWMAModelSettings,
  MovingAverageHoltModelSettings,
  MovingAverageModel,
  MovingAverageHoltWintersModelSettings,
  MovingAverage as SchemaMovingAverage,
} from './dataquery.gen';

export type MovingAverageModelSettings<T extends MovingAverageModel = MovingAverageModel> = Partial<
  Extract<
    | MovingAverageSimpleModelSettings
    | MovingAverageLinearModelSettings
    | MovingAverageEWMAModelSettings
    | MovingAverageHoltModelSettings
    | MovingAverageHoltWintersModelSettings,
    { model: T }
  >
>;

export interface MovingAverage<T extends MovingAverageModel = MovingAverageModel> extends SchemaMovingAverage {
  settings?: MovingAverageModelSettings<T>;
}
```